comet/broker.py

```diff
@@ -559,16 +470,20 @@ async def check_dataset(ds):
     to exist.
     """
     logger.debug("check_dataset: Checking dataset: {}".format(ds))
-    found = await wait_for_state(ds["state"])
-    if not found:
-        logger.debug("check_dataset: State of dataset unknown: {}".format(ds))
+    try:
+        await get_state(ds["state"])
+    except TimeoutError as err:
+        logger.debug("check_dataset: State of dataset {} unknown: {}".format(ds, err))
         return False
     if ds["is_root"]:
         logger.debug("check_dataset: dataset {} OK".format(ds))
         return True
-    found = await wait_for_dset(ds["base_dset"])
-    if not found:
-        logger.debug("check_dataset: Base dataset of dataset unknown: {}".format(ds))
+    try:
+        await get_dataset(ds["base_dset"])
+    except TimeoutError as err:
+        logger.debug(
+            "check_dataset: Base dataset of dataset {} unknown: {}".format(ds, err)
+        )
```

Comment:
Not that it's related to your change, but is there a reason this function doesn't give any debug logging output when it successfully checks a non-root dataset? Too much logging, or just that it got missed.